src/app/services/api-survey.submit.service.ts

```diff
@@ -0,0 +1,234 @@
+import { Injectable } from '@angular/core';
+import { Utilities } from '../common/utlilities';
+
+class PatientJSON {
+  patienID : string = '';
+  isDeceased : boolean = false;
+}
+
+class PatientSurveyJSON {
+    otoscopy : string = '';
+    typanometry : string = '';
+
+
+    rightEarLowFreqSeverity : string = '';
+    rightEarLowFreqConfiguration : string = '';
+    rightEarHighFreqSeverity : string = '';
+    rightEarHighFreqConfiguration : string = '';
+
+    leftEarLowFreqSeverity : string = '';
+    leftEarLowFreqConfiguration : string = '';
+    leftEarHighFreqSeverity : string = '';
+    leftEarHighFreqConfiguration : string = '';
+
+    audiogram : string = '';
+
+
+    thsSectionATotal : number = 0;
+    thsSectionBTotal : number = 0;
+    thsSectionCSeverity : number = 0;
+
+
+    tfiI : number = 0;
+    tfiSc : number = 0;
+    tfiC : number = 0;
+    tfiSi : number = 0;
+    tfiA : number = 0;
+    tfiR : number = 0;
+    tfiQ : number = 0;
+    tfiE : number = 0;
+
+    tfiOverallScore : number = 0;
+
+
+    tsTinnitusType : string = '';
+}
+
+class SurveyInstanceJSON {
+  patientSurvey : PatientSurveyJSON;
+  patient : PatientJSON;
+}
+
+@Injectable()
+export class SurveySubmitHandler {
+  // Note that anything passed in here is data that is not
+  // accessible via session storage.
+  public submitSurvey(thsScoreVars : Map<string, number>, tfiVars : Map<string, number>, tsType : string) : boolean {
+    if(thsScoreVars == null) {
+      console.log('SUBMISSION FAILED: NO THS SCORE DATA');
+      return false;
+    }
+
+    let testDataString = Utilities.getSessionStorage('tests-data');
+    let testData = JSON.parse(testDataString);
+
+    let result = new SurveyInstanceJSON;
```

Comment:
No constructor arguments?

Reply:
Yeah I'll fess up to it, I was getting lazy on this one. I'll tune it up to use proper constructors.